woodwork/deserialize.py

```diff
@@ -215,3 +182,97 @@ def _check_schema_version(saved_version_str):
+
+        ltype_metadata = col["logical_type"]
+        ltype = ww.type_system.str_to_logical_type(
+            ltype_metadata["type"], params=ltype_metadata["parameters"]
+        )
+
+        tags = col["semantic_tags"]
+
+        if "index" in tags:
+            tags.remove("index")
+        elif "time_index" in tags:
+            tags.remove("time_index")
+
+        logical_types[col_name] = ltype
+        semantic_tags[col_name] = tags
+        column_descriptions[col_name] = col["description"]
+        column_origins[col_name] = col["origin"]
+        column_metadata[col_name] = col["metadata"]
+        use_standard_tags[col_name] = col["use_standard_tags"]
+
+    return {
+        "name": typing_info.get("name"),
+        "index": typing_info.get("index"),
+        "time_index": typing_info.get("time_index"),
+        "logical_types": logical_types,
+        "semantic_tags": semantic_tags,
+        "column_descriptions": column_descriptions,
+        "column_origins": column_origins,
+        "table_metadata": typing_info.get("table_metadata"),
+        "column_metadata": column_metadata,
+        "use_standard_tags": use_standard_tags,
+    }
+
+
+def read_parquet(path, filename, lib="pandas", profile_name=None):
+    """
+    Read data from the parquet file at the location specified by ``path``
+    with the filename specified by ``filename``. Will initialize Woodwork using
+    the typing information stored in the parquet file metadata. If typing info
+    is not present, will initialize Woodwork without any arguments.
+
+    Args:
+        path (str): Location on disk to read from.
+        filename (str): Name of file to read. Will be ignored if reading dataframe with Dask.
+        lib (str): Name of library to use for reading data. Defaults to ``pandas``.
+            Should be one of ["pandas", "dask"]
+        profile_name (str) : Name of AWS profile to use, False to use an anonymous profile, or None.
+    """
+    import pyarrow as pa
+
+    if lib == "dask":
+        dataframe = dd.read_parquet(path)
+        # Read the metadata from the first parquet file in the directory
+        meta_filepath = glob.glob(os.path.join(path, "*.parquet"))[0]
+        file_metadata = pa.parquet.read_metadata(meta_filepath)
+    elif _is_s3(path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            local_filepath = os.path.join(tmpdir, filename)
+            s3_path = f"{path}/{filename}"
+            transport_params = None
+            transport_params = get_transport_params(profile_name)
+
+            use_smartopen(local_filepath, s3_path, transport_params)
+            dataframe = pd.read_parquet(local_filepath)
+            file_metadata = pa.parquet.read_metadata(local_filepath)
+    else:
+        filepath = os.path.join(path, filename)
+        dataframe = pd.read_parquet(filepath)
+        file_metadata = pa.parquet.read_metadata(filepath)
+
+    ww_metadata = file_metadata.metadata.get(str.encode(METADATA_KEY))
```

Comment:
Any reason we wouldn't just do this here to keep things consistent?
```suggestion
    ww_metadata = file_metadata.metadata.get(METADATA_KEY.encode())
```
Or maybe another thing we could do is just something like this:
```python
METADATA_KEY = b"_woodwork_metadata"
```
Since we always pass it to arrow which requires keys as bytes.

Reply:
No reason, that was just an oversight. I like the idea of updating `METADATA_KEY` to be bites from the start so we can avoid calling `.encode()` in several places.